src/server/scripts/Northrend/zone_borean_tundra.cpp

```diff
@@ -34,6 +34,138 @@
 #include "TemporarySummon.h"
 #include "WorldSession.h"
 
+ /*######
+ ## Quest 11590: Abduction
+ ######*/
+
+// NPC 25316: Beryl Sorcerer
+enum BerylSorcerer
+{
+    EVENT_FROSTBOLT                                = 1,
+    EVENT_ARCANE_CHAINS,
+
+    NPC_LIBRARIAN_DONATHAN                         = 25262,
+    NPC_CAPTURED_BERLY_SORCERER                    = 25474,
+
+    SPELL_FROSTBOLT                                = 9672,
+    SPELL_ARCANE_CHAINS                            = 45611,
+    SPELL_ARCANE_CHAINS_CHARACTER_FORCE_CAST       = 45625,
+    SPELL_ARCANE_CHAINS_SUMMON_CHAINED_MAGE_HUNTER = 45626,
+    SPELL_COSMETIC_ENSLAVE_CHAINS_SELF             = 45631,
+    SPELL_ARCANE_CHAINS_CHANNEL_II                 = 45735
+};
+
+struct npc_beryl_sorcerer : public CreatureAI
+{
+    npc_beryl_sorcerer(Creature* creature) : CreatureAI(creature), _chainsCast(false) { }
+
+    void JustAppeared() override
+    {
+        me->SetReactState(REACT_AGGRESSIVE);
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        if (me->IsValidAttackTarget(who))
+            AttackStart(who);
+
+        _events.ScheduleEvent(EVENT_FROSTBOLT, 3s, 4s);
+    }
+
+    void SpellHit(WorldObject* caster, SpellInfo const* spellInfo) override
+    {
+        if (_chainsCast)
+            return;
+
+        if (spellInfo->Id == SPELL_ARCANE_CHAINS)
+            if (Player* player = caster->ToPlayer())
+            {
+                _playerGUID = player->GetGUID();
+                _chainsCast = true;
+                _events.ScheduleEvent(EVENT_ARCANE_CHAINS, 4s);
+            }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
+
+        _events.Update(diff);
+
+        if (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+            case EVENT_FROSTBOLT:
+                DoCastVictim(SPELL_FROSTBOLT);
+                _events.ScheduleEvent(EVENT_FROSTBOLT, 3s, 4s);
+                break;
+            case EVENT_ARCANE_CHAINS:
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                {
+                    me->CastSpell(player, SPELL_ARCANE_CHAINS_CHARACTER_FORCE_CAST);
+                    player->KilledMonsterCredit(NPC_CAPTURED_BERLY_SORCERER);
+                }
+                me->DespawnOrUnsummon();
+                break;
+            default:
+                break;
+            }
+        }
+
+        DoMeleeAttackIfReady();
+    }
+private:
+    EventMap _events;
+    bool _chainsCast;
+    ObjectGuid _playerGUID;
+};
+
+// NPC 25474: Captured Beryl Sorcerer
+struct npc_captured_beryl_sorcerer : public FollowerAI
+{
+    npc_captured_beryl_sorcerer(Creature* creature) : FollowerAI(creature) { }
+
+    void JustAppeared() override
+    {
+        me->SetReactState(REACT_DEFENSIVE);
+        me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
```

Comment:
SetImmuneToAll please, without this flag
I know it doesn't have any of those flags but if hacking something, better to do it without that flag

Reply:
done